src/main/java/org/embulk/output/elasticsearch/ElasticsearchHttpClient.java

```diff
@@ -265,11 +266,43 @@ private void deleteIndex(String indexName, PluginTask task, Jetty92RetryHelper r
         // curl -XDELETE localhost:9200/{index}
         // Success: {"acknowledged":true}
         if (isIndexExisting(indexName, task, retryHelper)) {
+            getSnapshotStatusUntilDone(task, retryHelper);
             sendRequest(indexName, HttpMethod.DELETE, task, retryHelper);
             log.info("Deleted Index [{}]", indexName);
         }
     }
 
+    private void getSnapshotStatusUntilDone(PluginTask task, Jetty92RetryHelper retryHelper)
+    {
+        long execCount = 1;
+        long totalWaitingTime = 0;
+        while (isSnapshotProgressing(task, retryHelper)) {
+            try {
+                Thread.sleep(1000 * execCount);
+            }
+            catch (InterruptedException ex) {
+                // do nothing
+            }
+            if (execCount > 1) {
+                log.info("Waiting for snapshot completed.");
+            }
+            execCount++;
+            totalWaitingTime += 1000 * execCount;
+            if (totalWaitingTime > maxSnapshotWaitingTime) {
```

Comment:
Sorry I changed logic a80c72a


>  And please explain why you think that you don't take `RetryExecutor` provided by Embulk.

I thought RetryExecutor is too heavy for this purpose.
I just need exponential backoff mechanism but doesn't need exception handling or some others.

`isSnapshotProgressing` calls `sendRequest` and it uses [RetryExecutor](https://github.com/muga/embulk-output-elasticsearch/pull/36/files#diff-84310e25d97ec8051da42f94d72621f2R312)